sale_commission/models/account_move.py

```diff
@@ -0,0 +1,60 @@
+from odoo import models
+
+
+class AccountMove(models.Model):
+    _inherit = 'account.move'
+
+    def action_post(self):
+        """Overrides invoice confirmation to trigger commision calculation."""
+        res = super().action_post()
+        self._create_commission_lines('invoicing')
+        return res
+
+    def action_register_payment(self):
+        """Overrides payment registration to trigger commision calculation."""
+        res = super().action_register_payment()
+        self._create_commission_lines('payment')
+        return res
+
+    def _create_commission_lines(self, trigger_stage):
+        for invoice in self:
+            commission_rules = self.env['commission.rule'].search([])
```

Comment:
```suggestion
            commission_rules = self.env['commission.rule'].search([('due_at', '=', trigger_stage)])
```
We can fetch only needed records rather than fetching all the records